### PR TITLE
Add Markdown/Text Parser

### PR DESCRIPTION
**Pull request by yuvalyaron**
## Purpose

Added the ability to index Markdown and txt files.

Currently, since there is no Markdown viewer in the UI, when the user toggles the citation, the file is downloaded to their computer.

Part of #1365

## Does this introduce a breaking change?
```
[ ] Yes
[ X ] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[ X ] No
```

## Type of change

```
[ ] Bugfix
[ X ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](../CONTRIBUTING.md#submit-pr) for more details.

- [ X ] The current tests all pass (`python -m pytest`).
- [ X ] I added tests that prove my fix is effective or that my feature works
- [ X ] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ X ] I ran `python -m mypy` to check for type errors
- [ X ] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

File: data/Contoso_Electronics_Company_Overview.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Contoso and Sons
```
Comment (pamelafox) [inline data/Contoso_Electronics_Company_Overview.md:1]: Can you match the company name/type used in the PDFs?

File: scripts/prepdocslib/markdownparser.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    # match two or more newlines and replace them with one new line
+    output = re.sub(r"\n{2,}", "\n", data)
+    # match two or more spaces that are not newlines and replace them with one space
+    output = re.sub(r"[^\S\n]{2,}", " ", output)
+
+    return output.strip()
+
+
+class MarkdownParser(Parser):
```
Comment (pamelafox) [inline scripts/prepdocslib/markdownparser.py:23]: Does this currently do anything Markdown specific? I'm wondering if it should just be called a TextParser, and also be used for .txt formats. Did you check if azure-search-openai-javascript does any Markdown-specific processing?

File: data/Contoso_Electronics_Company_Overview.md (hunk 1 above)
Comment (yuvalyaron) [reply data/Contoso_Electronics_Company_Overview.md]: Done

File: scripts/prepdocslib/markdownparser.py (hunk 2 above)
Comment (yuvalyaron) [reply scripts/prepdocslib/markdownparser.py]: Thanks, naming it TextParser does make more sense, it's also similar to what `azure-search-openai-javascript` [have done](https://github.com/Azure-Samples/azure-search-openai-javascript/blob/main/packages/indexer/src/lib/formats/text.ts).

Comment (yuvalyaron) [reply scripts/prepdocslib/markdownparser.py]: I have also added a mapping for `.txt` files to use that parser as well

File: data/Contoso_Electronics_Company_Overview.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Recognizing the hard work and dedication of our employees is at the core of our culture. Here are some ways we celebrate achievements:
+
+- Monthly "Innovator of the Month" awards
+- Annual gala with awards for outstanding contributions
+- Team-building retreats for high-performing departments
+
+## Join Us!
+
+Contoso Electronics is always on the lookout for talented individuals who share our passion for innovation. If you're ready to be part of a dynamic team shaping the future of technology, check out our [careers page](https://careers.microsoft.com/) for exciting opportunities.
```
Comment (pamelafox) [inline data/Contoso_Electronics_Company_Overview.md:46]:
```suggestion
Contoso Electronics is always on the lookout for talented individuals who share our passion for innovation. If you're ready to be part of a dynamic team shaping the future of technology, check out our [careers page](http://www.contoso.com) for exciting opportunities.
```

File: data/Contoso_Electronics_Company_Overview.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+[Learn more about Contoso Electronics!](https://www.microsoft.com)
```
Comment (pamelafox) [inline data/Contoso_Electronics_Company_Overview.md:48]:
```suggestion
[Learn more about Contoso Electronics!](http://www.contoso.com)
```

**Review by pamelafox — APPROVED** [review]

**Pull request merged**